app/code/Magento/InventoryIndexer/Indexer/SelectBuilder.php

```diff
@@ -65,9 +78,15 @@ public function execute($stockId): Select
                 ['source_item' => $sourceItemTable],
                 [
                     SourceItemInterface::SKU,
-                    // Quick fix. Proper solution will be provided in https://github.com/magento-engcom/msi/issues/429
                     SourceItemInterface::QUANTITY => 'SUM(IF(source_item.' . SourceItemInterface::STATUS . ' = '
-                        . SourceItemInterface::STATUS_OUT_OF_STOCK . ', 0, IF(quantity > 0, quantity, 1)))',
+                        . SourceItemInterface::STATUS_OUT_OF_STOCK . ', 0, ' . SourceItemInterface::QUANTITY . '))',
+                    IndexStructure::IS_SALABLE => sprintf('MAX(IF(
+                    ((cataloginventory_stock_item.use_config_manage_stock = 1 AND %1$d = 1) OR (cataloginventory_stock_item.use_config_manage_stock = 0 AND cataloginventory_stock_item.manage_stock = 1))
```

Comment:
We can not work with table names directly
Need to resolve it via `$this->resourceConnection->getTableName`